### DISCUSSION TITLE
Skip Checkup Failure with Xcode 16 beta 3

### DISCUSSION THREAD

**Discussion by ygee07** [skiptools/skip#170] [General]
Attached Logs
[skip-checkup-2024-07-10T08:13:50Z.txt](https://github.com/user-attachments/files/16157772/skip-checkup-2024-07-10T08.13.50Z.txt)
[skip-doctor-2024-07-10T08:15:14Z.txt](https://github.com/user-attachments/files/16157775/skip-doctor-2024-07-10T08.15.14Z.txt)

**Comment by marcprux** [comment]
That's a curious error:

> error: couldn't build /private/var/folders/3z/jgz13lpn1wv7_xkhgp295sgh0000gn/T/2599C495-D934-4876-AD1B-185C616552B7/hello-skip/.build/plugins/outputs/skip-lib/SkipLib/destination/skipstone/.SkipLib.sourcehash because of missing inputs: /private/var/folders/3z/jgz13lpn1wv7_xkhgp295sgh0000gn/T/2599C495-D934-4876-AD1B-185C616552B7/hello-skip/.build/plugins/outputs/skip-lib/skip-unit/SkipUnit/skipstone/.SkipUnit.sourcehash

Could you try running `skip checkup` a second time and see if the error is the same? And in addition to attaching the build log, can you also paste the terminal output of the checkup command so we can see the reported summary?

**Reply by ygee07** [reply]
Here's the latest run

<img width="976" alt="Screenshot 2024-07-10 at 9 14 19 PM" src="https://github.com/skiptools/skip/assets/2621883/94955feb-f0f2-4260-843c-71d1dc1250f2">

[skip-checkup-2024-07-10T13:12:48Z.txt](https://github.com/user-attachments/files/16162628/skip-checkup-2024-07-10T13.12.48Z.txt)

**Reply by marcprux** [reply]
Thanks for sending. I fear this may be an issue with Swift 6 and/or Xcode 16. Are you running Xcode 16 Beta 3, or an earlier version?

Do you happen to still have Xcode 15 on your machine (which comes with Swift 5.10)? If so, could you try setting that as your current Xcode, either with `xcode-select` or just by replacing the Xcode binary in /Applications/ and re-running `skip checkup`? I would be interested in seeing if it passes.

**Reply by marcprux** [reply]
Update: it has been reported that this breakage is new in Xcode 16 beta 3. Xcode 16 beta 2 works.

**Reply by ygee07** [reply]
Thanks! This would explain why I'm just encountering it now. Didn't have any issues before.

**Reply by marcprux** [reply]
You might also try running your project in Xcode 16 beta 3 (or creating a new one with `skip init --appid=… project-name AppTitle`). It may be the case that it is just `skip checkup` that is failing, as we have other reports of people being able to run their projects from Xcode.

**Reply by ygee07** [reply]
There were errors when I did the init command (probably same errors reported in the checkup), but it created the project and I was able to build it without problems (the most important thing).

Just figured I let the community be aware of the skip checkup issue. But no problems aside from that.
